src/apidoc.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 MOSEC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::BTreeMap, str::FromStr};
+
+use hyper::StatusCode;
+use serde::Deserialize;
+use utoipa::openapi::{
+    path::Operation, request_body::RequestBody, Components, Content, ContentBuilder, OpenApi,
+    PathItemType, RefOr, ResponseBuilder, Responses, Schema,
+};
+
+#[derive(Deserialize, Default)]
+pub(crate) struct InferenceSchemas {
+    req_schema: RefOr<Schema>,
+    res_schema: RefOr<Schema>,
+    schemas: BTreeMap<String, RefOr<Schema>>,
+}
+
+impl FromStr for InferenceSchemas {
+    type Err = serde_json::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str::<InferenceSchemas>(s)
+    }
+}
+
+pub(crate) struct MosecApiDoc {
+    pub rust_api: OpenApi,
+}
+
+impl MosecApiDoc {
+    fn get_operation<'a>(
+        &self,
+        api: &'a mut OpenApi,
+        route: &str,
+        method: &PathItemType,
+    ) -> Option<&'a mut Operation> {
+        let path = api.paths.paths.get_mut(route).unwrap();
+        path.operations.get_mut(method)
+    }
+
+    fn get_route_request_body<'a>(
+        &self,
+        api: &'a mut OpenApi,
+        route: &str,
+        method: &PathItemType,
+    ) -> Option<&'a mut RequestBody> {
+        let op = self.get_operation(api, route, method).unwrap();
+        op.request_body.as_mut()
+    }
+
+    fn get_route_responses<'a>(
```

Comment:
Is it necessary to have an explicit lifetime?

Reply:
Actually I annotated lifetime on more functions which ware removed under clippy's warning. 
I didn't really understand this part, will continue to learn.